File: source/services/shared/ev-core/src/charger.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Charger {
    pub id: String,
    pub station_id: String,
    #[serde(rename = "type")]
    pub charger_type: String,
    pub power_kw: f64,
    pub status: String,
    pub price_per_kwh: f64,
    pub created_at: chrono::NaiveDateTime,
    pub updated_at: chrono::NaiveDateTime,
    pub deleted_at: Option<chrono::NaiveDateTime>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateChargerRequest {
    #[serde(rename = "type")]
    pub charger_type: String,
    pub power_kw: f64,
    pub price_per_kwh: Option<f64>,
}

File: apps/admin-dashboard/src/types/charger.ts
import type { BaseEntity } from './common';

export interface Charger extends BaseEntity {
  station_id: string;
  power_rating: number;
  deleted_at: string | null;
}

export interface CreateChargerRequest {
  station_id: string;
  status: string;
  power_rating: number;
}
